.github/workflows/ci_cd.yml

```diff
@@ -79,6 +78,42 @@ jobs:
           command: node -e "process.exit(1)"
           on_retry_command: node -e "console.log('this is a retry command')"
 
+      - name: retry_on_exit_code (with expected error code)
+        id: retry_on_exit_code_expected
+        uses: ./
+        continue-on-error: true
+        with:
+          timeout_minutes: 1
+          retry_on_exit_code: 2
+          max_attempts: 3
+          command: node -e "process.exit(2)"
+      - uses: nick-invision/assert-action@v1
+        with:
+          expected: failure
+          actual: ${{ steps.retry_on_exit_code_expected.outcome }}
+      - uses: nick-invision/assert-action@v1
+        with:
+          expected: 3
+          actual: ${{ steps.retry_on_exit_code_expected.outputs.total_attempts }}
+
+      - name: retry_on_exit_code (with unexpected error code)
```

Comment:
@andersfischernielsen @erwinw

Can you take a look at [this](https://github.com/nick-fields/retry/runs/6077868023?check_suite_focus=true#step:19:32) test failure?  I made a few tweaks to your tests, first being that a duplicate `id` was used which caused the entire workflow to fail, then I added assertions to your two tests.

The assertion is failing for your `retry_on_exit_code (with unexpected error code)` test.  I assert that if an unexpected exit code is thrown (exit code 1), and the `retry_on_exit_code` is set (exit code 2), then the rerun should only occur once not 3x as is happening currently.  I _think_ you need to also handle `max_attempts`  when `retry_on_exit_code` is set and skip additional attempts if an unexpected exit code is returned.

Reply:
Hey @nick-fields , well danged, I've ~poked~ gently asked @andersfischernielsen to have a look at this!